src/creator.rs

```diff
@@ -76,18 +73,17 @@ impl<H: Hasher> Creator<H> {
         let control_hash = ControlHash::new(&parents);
 
         let new_preunit = PreUnit::new(self.node_ix, round, control_hash);
-        trace!(target: "AlephBFT-creator", "{:?} Created a new unit {:?} at round {:?}.", self.node_ix, new_preunit, self.current_round);
+        trace!(target: "AlephBFT-creator", "{:?} Created a new unit {:?} at round {:?}.", self.node_ix, new_preunit, round);
         self.new_units_tx
             .unbounded_send(NotificationOut::CreatedPreUnit(new_preunit))
             .expect("Notification channel should be open");
 
-        self.current_round += 1;
-        self.init_round(self.current_round);
+        self.init_round(round + 1);
     }
 
     fn add_unit(&mut self, round: Round, pid: NodeIndex, hash: H::Hash) {
         // units that are too old are of no interest to us
-        if round + 1 >= self.current_round {
+        if usize::from(round + 1) >= self.n_candidates_by_round.len() - 1 {
```

Comment:
Elsewhere in the code we use `as usize` for this, perhaps change for consistency?

Reply:
There are a couple of instances of this.

Reply:
Done